### PR TITLE
Use official Timely, Differential, Columnation

### PR DESCRIPTION
**Pull request by antiguru**
We've resumed publishing Timely, Differential, and Columnation to crates, which allows us to switch away from our own forks of said projects.

### Checklist

- [ ] This PR has adequate test coverage / QA involvement has been duly considered. ([trigger-ci for additional test/nightly runs](https://trigger-ci.dev.materialize.com/))
- [ ] This PR has an associated up-to-date [design doc](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/README.md), is a design doc ([template](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/00000000_template.md)), or is sufficiently small to not require a design.
  
- [ ] If this PR evolves [an existing `$T ⇔ Proto$T` mapping](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/command-and-response-binary-encoding.md) (possibly in a backwards-incompatible way), then it is tagged with a `T-proto` label.
- [ ] If this PR will require changes to cloud orchestration or tests, there is a companion cloud PR to account for those changes that is tagged with the release-blocker label ([example](https://github.com/MaterializeInc/cloud/pull/5021)).
  
- [ ] If this PR includes major [user-facing behavior changes](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/guide-changes.md#what-changes-require-a-release-note), I have pinged the relevant PM to schedule a changelog post.

### CODE REVIEW

File: misc/cargo-vet/audits.toml [hunk 1]
```diff
@@ -1513,7 +1483,7 @@ delta = "6.0.6 -> 7.0.0"
 
 [[trusted.abomonation]]
 criteria = "safe-to-deploy"
-user-id = 704
+user-id = 704 # Frank McSherry (frankmcsherry)
```
Comment (frankmcsherry) [inline misc/cargo-vet/audits.toml:1486]: Are we still using abomonation for anything load bearing, and how can we fix that?

Comment (antiguru) [reply misc/cargo-vet/audits.toml:1486]: Nope! https://github.com/MaterializeInc/materialize/pull/30264 to remove it.

**Review by frankmcsherry — APPROVED** [review]
I don't understand the `cargo vet` stuff, but I can be filled in async if it is load-bearing. Otherwise, seems good.

**Comment by antiguru** [comment]
The `cargo vet` changes are needed because we don't depend on git revisions anymore. Additionally, we generally trust the columnation/timely/differential dependencies without needing to update for every new upstream version, which was always the intention.

**Pull request merged**